src/client/__tests__/EmailInput.test.tsx

```diff
@@ -0,0 +1,81 @@
+/**
```

Comment:
loving these tests, i wonder if we need to spend some time adding them in for existing components too?

Reply:
they're fun to write :) yes! Components like `PasswordInput` with a bit of state might be good candidates. I could put together a list

Reply:
even without state, they're useful in a lot of cases tbh

Reply:
Maybe a good dev standup topic!